core/domain/translation_services.py

```diff
@@ -56,13 +55,19 @@ def get_and_cache_machine_translation(
     if translation is not None:
         return translation.translated_text
 
-    translated_text = cloud_translate_services.translate_text(
-        source_text, source_language_code, target_language_code)
+    translated_text = None
+    try:
+        translated_text = cloud_translate_services.translate_text(
+            source_text, source_language_code, target_language_code)
+    except ValueError as e: # Language code not allowlisted
+        logging.error(e)
+
+    if translated_text is not None:
```

Comment:
There should be a test that verifies that no model gets created for "the translated_text is not None" case. Always try to test any new behaviour and the different branches; line coverage alone is insufficient.

Reply:
Thanks for catching this! Done.